### PR TITLE
feat(cb2-3411): added seatbelts to test form

### PR DESCRIPTION
**Pull request by kpearmanbjss**
## View Seatbelt installation check section for a PSV test record

Added the seatbelt fields to the Test results Form. 
[CB2-3411](https://dvsa.atlassian.net/browse/CB2-3411)

## Checklist

- [ ] Branch is rebased against the latest develop/common
- [ ] Necessary `id` required prepended with `"test-"` have been checked with automation testers and added
- [ ] Code and UI has been tested manually after the additional changes
- [ ] PR title includes the JIRA ticket number
- [ ] Squashed commits contain the JIRA ticket number
- [ ] Link to the PR added to the repo
- [ ] Delete branch after merge

### CODE REVIEW

**Review by gjulien-bjss — APPROVED** [review]

File: src/app/models/test-type.model.ts [hunk 1]
```diff
@@ -15,4 +15,8 @@ export interface TestType {
   prohibitionIssued: boolean;
 
   testResult: string;
+
+  seatbeltInstallationCheckDate: boolean;
```
Comment (Ian-Hodges) [inline src/app/models/test-type.model.ts:19]: Should seatbeltInstallationCheckDate have 'Date' at the end? Sounds wrong when it's a boolean.

File: src/app/models/test-type.model.ts [hunk 2]
```diff
@@ -15,4 +15,8 @@ export interface TestType {
...
+  numberOfSeatbeltsFitted: number;
+  lastSeatbeltInstallationCheckDate: string | Date;
```
Comment (danielchildsbjss) [inline src/app/models/test-type.model.ts:21]: Is this the correct type? Feels like it should be a nullable Date

**Review by naathanbrown — APPROVED** [review]

File: src/app/models/test-type.model.ts (hunk 2 above)
Comment (kpearmanbjss) [reply src/app/models/test-type.model.ts:21]: I went off of what the other fields were using, I don't think we've had something that uses a nullable date yet so not entirely sure how I would change it to match that?

File: src/app/models/test-type.model.ts (hunk 1 above)
Comment (kpearmanbjss) [reply src/app/models/test-type.model.ts:19]: I was also confused, that is the way it's laid out on the DB according to the spreadsheet I was given, it's a boolean called seatbeltInstallationCheckDate, it shows whether or not the seatbelt checks were carried out during this test or not.

Comment (Ian-Hodges) [reply src/app/models/test-type.model.ts:19]: Sounds like it is a boolean then, just badly named.

File: src/app/models/test-type.model.ts (hunk 2 above)
Comment (kpearmanbjss) [reply src/app/models/test-type.model.ts]: Looking at the data, the field appears to be "Were the seatbelts checked on this date" and if so it makes the other fields mandatory. I think lastSeatbeltInstallationCheckDate is a literal "The last time we checked the seatbelts".

I've gone on the assumption by the way these need to match the structure of how they are in the DB? If not then i'd happily rename it to something more useful but not sure how i'd point it to the right field

File: src/app/forms/templates/psv/psv-annual.template.ts [hunk 3]
```diff
@@ -222,6 +222,34 @@ export const PsvAnnual: FormNode = {
...
+              name: 'numberOfSeatbeltsFitted',
+              label: 'Number of seatbelts fitted',
+              value: '',
+              children: [],
+              type: FormNodeTypes.CONTROL
+            },
+            {
+              name: 'lastSeatbeltInstallationCheckDate',
+              label: 'Most recent installation check',
+              value: '',
```
Comment (danielchildsbjss) [inline src/app/forms/templates/psv/psv-annual.template.ts:249]: I think then this also becomes null, if the control supports that

**Review by Ian-Hodges — APPROVED** [review]

**Pull request merged**